
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      <link rel="icon" href="../../assets/images/SIB_logo.svg">
      <meta name="generator" content="mkdocs-1.2.3, mkdocs-material-8.2.5">
    
    
      
        <title>Group work - Long-read sequencing analysis</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.2d9f7617.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.e6a45f82.min.css">
        
      
    
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../stylesheets/extra.css">
    
    <script>__md_scope=new URL("../..",location),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="" data-md-color-primary="none" data-md-color-accent="none">
  
    
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#learning-outcomes" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
      <div data-md-component="outdated" hidden>
        <aside class="md-banner md-banner--warning">
          
        </aside>
      </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="Long-read sequencing analysis" class="md-header__button md-logo" aria-label="Long-read sequencing analysis" data-md-component="logo">
      
  <img src="../../assets/images/SIB_logo.svg" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Long-read sequencing analysis
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Group work
            
          </span>
        </div>
      </div>
    </div>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/sib-swiss/NGS-longreads-training/" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    sib-swiss/NGS-longreads-training
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="Long-read sequencing analysis" class="md-nav__button md-logo" aria-label="Long-read sequencing analysis" data-md-component="logo">
      
  <img src="../../assets/images/SIB_logo.svg" alt="logo">

    </a>
    Long-read sequencing analysis
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/sib-swiss/NGS-longreads-training/" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    sib-swiss/NGS-longreads-training
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        Home
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../precourse/" class="md-nav__link">
        Precourse preparations
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../course_schedule/" class="md-nav__link">
        Course schedule
      </a>
    </li>
  

    
      
      
      

  
  
    
  
  
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_4" type="checkbox" id="__nav_4" checked>
      
      
      
      
        <label class="md-nav__link" for="__nav_4">
          Course material
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Course material" data-md-level="1">
        <label class="md-nav__title" for="__nav_4">
          <span class="md-nav__icon md-icon"></span>
          Course material
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../introduction/" class="md-nav__link">
        Introduction
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../server_login/" class="md-nav__link">
        Server login
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../qc_alignment/" class="md-nav__link">
        QC and alignment
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../applications/" class="md-nav__link">
        Applications
      </a>
    </li>
  

            
          
            
              
  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" data-md-toggle="toc" type="checkbox" id="__toc">
      
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          Group work
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        Group work
      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#learning-outcomes" class="md-nav__link">
    Learning outcomes
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#material" class="md-nav__link">
    Material
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#introduction" class="md-nav__link">
    Introduction
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#roles-organisation" class="md-nav__link">
    Roles &amp; organisation
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#working-directories" class="md-nav__link">
    Working directories
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#project-1-differential-isoform-expression-analysis-of-ont-data" class="md-nav__link">
     Project 1: Differential isoform expression analysis of ONT data
  </a>
  
    <nav class="md-nav" aria-label=" Project 1: Differential isoform expression analysis of ONT data">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#before-you-start" class="md-nav__link">
    Before you start
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#tasks-questions" class="md-nav__link">
    Tasks &amp; questions
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#project-2-repeat-expansion-analysis-of-pacbio-data" class="md-nav__link">
     Project 2: Repeat expansion analysis of PacBio data
  </a>
  
    <nav class="md-nav" aria-label=" Project 2: Repeat expansion analysis of PacBio data">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#before-you-start_1" class="md-nav__link">
    Before you start
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#tasks-questions_1" class="md-nav__link">
    Tasks &amp; questions
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#learning-outcomes" class="md-nav__link">
    Learning outcomes
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#material" class="md-nav__link">
    Material
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#introduction" class="md-nav__link">
    Introduction
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#roles-organisation" class="md-nav__link">
    Roles &amp; organisation
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#working-directories" class="md-nav__link">
    Working directories
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#project-1-differential-isoform-expression-analysis-of-ont-data" class="md-nav__link">
     Project 1: Differential isoform expression analysis of ONT data
  </a>
  
    <nav class="md-nav" aria-label=" Project 1: Differential isoform expression analysis of ONT data">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#before-you-start" class="md-nav__link">
    Before you start
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#tasks-questions" class="md-nav__link">
    Tasks &amp; questions
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#project-2-repeat-expansion-analysis-of-pacbio-data" class="md-nav__link">
     Project 2: Repeat expansion analysis of PacBio data
  </a>
  
    <nav class="md-nav" aria-label=" Project 2: Repeat expansion analysis of PacBio data">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#before-you-start_1" class="md-nav__link">
    Before you start
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#tasks-questions_1" class="md-nav__link">
    Tasks &amp; questions
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content" data-md-component="content">
            <article class="md-content__inner md-typeset">
              
                
<a href="https://github.com/sib-swiss/NGS-longreads-training/edit/master/docs/course_material/group_work.md" title="Edit this page" class="md-content__button md-icon">
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25z"/></svg>
</a>



  <h1>Group work</h1>

<h2 id="learning-outcomes">Learning outcomes</h2>
<p><strong>After having completed this chapter you will be able to:</strong></p>
<ul>
<li>Develop a basic pipeline for alignment-based analysis of a long-read sequencing dataset</li>
<li>Answer biological questions based on the analysis resulting from the pipeline</li>
</ul>
<h2 id="material">Material</h2>
<p><a class="md-button" href="../../assets/pdf/group_work.pdf"><span class="twemoji"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 384 512"><!--! Font Awesome Free 6.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M184 208c0-4.406-3.594-8-8-8s-8 3.6-8 8c0 2.062.297 23.31 9.141 50.25C179.1 249.6 184 226.2 184 208zM256 0v128h128L256 0zM80 422.4c0 9.656 10.47 11.97 14.38 6.375C99.27 421.9 108.8 408 120.1 388.6c-14.22 7.969-27.25 17.31-38.02 28.31-1.33 1.39-2.08 3.39-2.08 5.49zM224 128V0H48C21.49 0 0 21.49 0 48v416c0 26.5 21.49 48 48 48h288c26.51 0 48-21.49 48-48V160H256.9c-18.6 0-32.9-14.3-32.9-32zm68 184c24.26 0 44 19.74 44 44 0 24.67-18.94 44-43.13 44-5.994 0-11.81-.953-17.22-2.805-20.06-6.758-38.38-15.96-54.55-27.39-23.88 5.109-45.46 11.52-64.31 19.1-14.43 26.31-27.63 46.15-36.37 58.41C112.1 457.8 100.8 464 87.94 464 65.92 464 48 446.1 48 424.1c0-11.92 3.74-21.82 11.18-29.51 16.18-16.52 37.37-30.99 63.02-43.05 11.75-22.83 21.94-46.04 30.33-69.14-16.33-40-16.53-74-16.53-74.4 0-22.05 17.95-40 40-40 22.06 0 40 17.95 40 40 0 24.1-7.227 55.75-8.938 62.63a587.512 587.512 0 0 1-3.082 9.723c7.83 14.46 17.7 27.21 29.44 38.05C263.1 313.4 284.3 312.1 287.6 312h4.4zm-135.5 42.6c17.98-6.5 36.13-11.44 52.92-15.19-12.42-12.06-22.17-25.12-29.8-38.16a617.231 617.231 0 0 1-23.12 53.35zM292.9 368c6.1 0 11.1-5 11.1-11.1 0-7.5-5.4-12.9-12-12.9h-4c-.344.031-16.83.969-40.95 4.75 11.27 7 24.12 13.19 38.84 18.12 2.11.73 4.61 1.13 7.01 1.13z"/></svg></span> Download the presentation</a></p>
<h2 id="introduction">Introduction</h2>
<p>The last part of this course will consist of project-based-learning. This means that you will work in groups on a single question. We will split up into groups of five people.</p>
<div class="admonition note">
<p class="admonition-title">If working independently</p>
<p>If you are working independently, you probably can not work in a group. However, you can test your skills with these real biological datasets. Realize that the datasets and calculations are (much) bigger compared to the exercises, so check if your computer is up for it. You&rsquo;ll probably need around 4 cores, 16G of RAM and 10G of harddisk.</p>
</div>
<div class="admonition note">
<p class="admonition-title">If online</p>
<p>If the course takes place online, we will use break-out rooms to communicate within groups. Please stay in the break-out room during the day, also if you are working individually.</p>
</div>
<h2 id="roles-organisation">Roles &amp; organisation</h2>
<p>Project based learning is about learning by doing, but also about <em>peer instruction</em>. This means that you will be both a learner and a teacher. There will be differences in levels among participants, but because of that, some will learn efficiently from people that have just learned, and others will teach and increase their understanding.</p>
<p>Each project has <strong>tasks</strong> and <strong>questions</strong>. By performing the tasks, you should be able to answer the questions. At the start of the project, make sure that each of you gets a task assigned. You should consider the tasks and questions as a guidance. If interesting questions pop up during the project, you are <strong>encouraged</strong> to work on those. Also, you don&rsquo;t have to perform all the tasks and answer all the questions.</p>
<p>In the afternoon of day 1, you will divide the initial tasks, and start on the project. On day 2, you can work on the project in the morning and in the first part of the afternoon. We will conclude the projects with a <strong>10-minute presentation</strong> of each group.</p>
<h2 id="working-directories">Working directories</h2>
<p>Each group has access to a shared working directory. It is mounted in the root directory (<code>/</code>). Make a soft link in your home directory:</p>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span> ~
ln -s /group_work/&lt;group name&gt; ./
</code></pre></div>
<p>Now you can find your group directory at <code>~/&lt;group name&gt;</code>. Use this as much as possible.</p>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>Do not remove the soft link with <code>rm -r</code>, this will delete the entire source directory. If you want to remove only the softlink, use <code>rm</code> (without <code>-r</code>), or <code>unlink</code>. More info <a href="https://linuxize.com/post/how-to-remove-symbolic-links-in-linux/">here</a>.</p>
</div>
<h2 id="project-1-differential-isoform-expression-analysis-of-ont-data"><span class="twemoji"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M184 0c30.9 0 56 25.07 56 56v400c0 30.9-25.1 56-56 56-28.9 0-52.7-21.9-55.7-50.1-5.2 1.4-10.7 2.1-16.3 2.1-35.35 0-64-28.7-64-64 0-7.4 1.27-14.6 3.59-21.2C21.43 367.4 0 338.2 0 304c0-31.9 18.71-59.5 45.77-72.3C37.15 220.8 32 206.1 32 192c0-30.7 21.59-56.3 50.41-62.6C80.84 123.9 80 118 80 112c0-29.94 20.6-55.08 48.3-62.07C131.3 21.86 155.1 0 184 0zm199.7 49.93C411.4 56.92 432 82.06 432 112c0 6-.8 11.9-2.4 17.4 28.8 6.3 50.4 31.9 50.4 62.6 0 14.1-5.1 28.8-13.8 39.7 27.1 12.8 45.8 40.4 45.8 72.3 0 34.2-21.4 63.4-51.6 74.8 2.3 6.6 3.6 13.8 3.6 21.2 0 35.3-28.7 64-64 64-5.6 0-11.1-.7-16.3-2.1-3 28.2-26.8 50.1-55.7 50.1-30.9 0-56-25.1-56-56V56c0-30.93 25.1-56 56-56 28.9 0 52.7 21.86 55.7 49.93z"/></svg></span> Project 1: Differential isoform expression analysis of ONT data</h2>
<p>In this project, you will be working with data from the same resource as the data we have already worked on:</p>
<p>Clark, M. B. et al (2020). <em>Long-read sequencing reveals the complex splicing profile of the psychiatric risk gene CACNA1C in human brain</em>. Molecular Psychiatry, 25(1), 37–47. <a href="https://doi.org/10.1038/s41380-019-0583-1">https://doi.org/10.1038/s41380-019-0583-1</a>.</p>
<p>It is Oxford Nanopore Technology sequencing data of PCR amplicons of the gene CACNA1C. It is primarily used to discover new splice variants. We will use the dataset to do that and in addition do a differential isoform expression analysis with <a href="https://github.com/BrooksLabUCSC/flair">FLAIR</a>.</p>
<div class="admonition info">
<p class="admonition-title">Project aim</p>
<p>Discover new splice variants and identify differentially expressed isoforms.</p>
</div>
<p>You can download the required data like this:</p>
<div class="highlight"><pre><span></span><code>wget https://ngs-longreads-training.s3.eu-central-1.amazonaws.com/groupwork_ont.tar.gz
tar -xvf groupwork_ont.tar.gz
rm groupwork_ont.tar.gz
</code></pre></div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Download the data file package in your shared working directory, i.e. : <code>/group_work/&lt;group name&gt;</code> or <code>~/&lt;group name&gt;</code>. Only one group member has to do this.</p>
</div>
<p>This will create a directory <code>groupwork_ont</code> with the following structure:</p>
<div class="highlight"><pre><span></span><code>groupwork_ont/
├── alignments
│   ├── cerebellum-5238-batch2.bam
│   ├── cerebellum-5298-batch2.bam
│   ├── cerebellum-5346-batch2.bam
│   ├── parietal_cortex-5238-batch1.bam
│   ├── parietal_cortex-5298-batch1.bam
│   └── parietal_cortex-5346-batch1.bam
├── counts
│   └── counts_matrix_test.tsv
├── reads
│   ├── cerebellum-5238-batch2.fastq.gz
│   ├── cerebellum-5298-batch2.fastq.gz
│   ├── cerebellum-5346-batch2.fastq.gz
│   ├── parietal_cortex-5238-batch1.fastq.gz
│   ├── parietal_cortex-5298-batch1.fastq.gz
│   ├── parietal_cortex-5346-batch1.fastq.gz
│   ├── striatum-5238-batch2.fastq.gz
│   ├── striatum-5298-batch2.fastq.gz
│   └── striatum-5346-batch2.fastq.gz
├── reads_manifest.tsv
└── scripts
    └── differential_expression_example.Rmd

4 directories, 18 files
</code></pre></div>
<p>Download the fasta file and gtf like this:</p>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span> groupwork_ont/
mkdir reference
<span class="nb">cd</span> reference
wget ftp://ftp.ensembl.org/pub/release-102/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.chromosome.12.fa.gz
wget ftp://ftp.ensembl.org/pub/release-102/gtf/homo_sapiens/Homo_sapiens.GRCh38.102.gtf.gz
gunzip *.gz
</code></pre></div>
<h3 id="before-you-start">Before you start</h3>
<p>You can start this project with dividing initial tasks. Because some intermediate files are already given, participants can develop scripts/analyses at different steps of the full analysis from the start. Possible starting points are:</p>
<ul>
<li>Quality control, running <code>fastqc</code> and <code>NanoPlot</code></li>
<li>Alignment, running <code>minimap2</code></li>
<li>Develop scripts required to run FLAIR</li>
<li>Differential expression analysis.</li>
</ul>
<h3 id="tasks-questions">Tasks &amp; questions</h3>
<ul>
<li>Perform QC with <code>fastqc</code> and with <code>NanoPlot</code>.<ul>
<li>Do you see a difference between them?</li>
<li>How is the read quality compared to the publication?</li>
</ul>
</li>
<li>Align each sample separately with <code>minimap2</code> with default parameters. Set parameters <code>-x</code> and <code>-G</code> to the values we have used during the <a href="../qc_alignment#3-read-alignment">QC and alignment exercises</a>. You can use 4 threads (set the number of threads with <code>-t</code>)</li>
</ul>
<div class="admonition danger">
<p class="admonition-title">Start the alignment as soon as possible</p>
<p>The alignment takes about 6 minutes per sample, so in total about one hour to run. Try to start the alignment as soon as possible. You can speed up your alignment by first making an index, e.g.:</p>
<div class="highlight"><pre><span></span><code>minimap2 <span class="se">\</span>
-x splice <span class="se">\</span>
-d reference/Homo_sapiens.GRCh38.dna.chromosome.12.fa.mmi <span class="se">\</span>
reference/Homo_sapiens.GRCh38.dna.chromosome.12.fa
</code></pre></div>
<p>Refer to the generated index (<code>.mmi</code> file) as reference in the alignment command, e.g.:</p>
<div class="highlight"><pre><span></span><code>minimap2 \
-a \
-x splice \
-G 500k \
-t 4 \
reference/Homo_sapiens.GRCh38.dna.chromosome.12.fa.mmi \
reads/&lt;my_reads.fastq.gz&gt;
</code></pre></div>
</div>
<ul>
<li>Clone the <a href="https://github.com/BrooksLabUCSC/flair">FLAIR repository</a> to the server, and check out the documentation. All FLAIR dependencies are in the the pre-installed conda environment named <code>flair</code>. You can activate it with <code>conda activate flair</code>.</li>
<li>Merge the separate alignments with <code>samtools merge</code>, index the merged bam file, and generate a <code>bed12</code> file with the script <code>flair/bin/bam2Bed12.py</code></li>
<li>Run <code>flair.py correct</code> on the <code>bed12</code> file. Add the <code>gtf</code> to the options to improve the alignments.</li>
<li>Run <code>flair.py collapse</code> to generate isoforms from corrected reads. This steps takes ~1 hour to run.</li>
<li>Generate a count matrix with <code>flair.py quantify</code> by using the isoforms fasta and <code>reads_manifest.tsv</code>.</li>
</ul>
<div class="admonition danger">
<p class="admonition-title">Paths in <code>reads_manifest.tsv</code></p>
<p>The paths in <code>reads_manifest.tsv</code> are relative, e.g. <code>reads/striatum-5238-batch2.fastq.gz</code> points to a file relative to the directory from which you are running <code>flair.py quantify</code>. So the directory from which you are running the command should contain the directory <code>reads</code>. If not, modify the paths in the file accordingly (use full paths if you are not sure).</p>
</div>
<ul>
<li>Now you can do several things:<ul>
<li>Do a differential expression analysis. In <code>scripts/</code> there&rsquo;s a basic R script to do the analysis. Go to your specified IP and port to login to RStudio server (the username is <code>rstudio</code>).</li>
<li>Investigate the isoform usage with the flair script <code>plot_isoform_usage.py</code></li>
<li>Investigate productivity of the different isoforms.</li>
</ul>
</li>
</ul>
<h2 id="project-2-repeat-expansion-analysis-of-pacbio-data"><span class="twemoji"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 512 512"><!--! Font Awesome Free 6.0.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="m472.2 195.9-66.1-22.1c-19.25-6.624-33.5-20.87-38.13-38.24l-16-60.49c-11.62-43.74-76.63-57.11-110-22.62L194.1 99.3c-13.25 13.75-33.5 20.87-54.25 19.25l-70.99-5.65c-52-3.999-86.88 44.99-59 82.86l38.63 52.49c11 14.1 12.75 33.74 4.625 50.12l-28.5 56.99c-20.62 41.24 22.88 84.86 73.5 73.86l69.1-15.25c20.12-4.499 41.38 0 57 11.62l54.38 40.87c39.38 29.62 101 7.623 104.5-37.24l4.625-61.86c1.375-17.75 12.88-33.87 30.62-42.99l61.1-31.62c46.66-22.95 41.46-80.25-7.24-96.85zM159.1 256c-17.62 0-31.1-14.37-31.1-31.1s14.37-31.1 31.1-31.1 31.1 14.37 31.1 31.1-12.6 31.1-31.1 31.1zm128 95.1c-17.62 0-31.1-14.37-31.1-31.1 0-17.62 14.37-31.1 31.1-31.1s31.1 14.37 31.1 31.1c.9 17.6-12.6 31.1-31.1 31.1zm16-127.1c-8.875 0-15.1-7.125-15.1-15.1 0-8.873 7.125-15.1 15.1-15.1s15.1 7.125 15.1 15.1c.9 8-5.3 15.1-15.1 15.1z"/></svg></span> Project 2: Repeat expansion analysis of PacBio data</h2>
<p>You will be working with data from an experiment in which DNA of 8 individuals was sequenced for five different targets by using Pacbio&rsquo;s no-Amp targeted sequencing system. Two of these targets contain repeat expansions that are related to a disease phenotype.</p>
<div class="admonition info">
<p class="admonition-title">Project aim</p>
<p>Estimate variation in repeat expansions in two target regions, and relate them to a disease phenotype.</p>
</div>
<table>
<thead>
<tr>
<th>individual</th>
<th>disease1</th>
<th>disease2</th>
</tr>
</thead>
<tbody>
<tr>
<td>1015</td>
<td>disease</td>
<td>healthy</td>
</tr>
<tr>
<td>1016</td>
<td>disease</td>
<td>healthy</td>
</tr>
<tr>
<td>1017</td>
<td>disease</td>
<td>healthy</td>
</tr>
<tr>
<td>1018</td>
<td>disease</td>
<td>healthy</td>
</tr>
<tr>
<td>1019</td>
<td>healthy</td>
<td>healthy</td>
</tr>
<tr>
<td>1020</td>
<td>healthy</td>
<td>disease</td>
</tr>
<tr>
<td>1021</td>
<td>healthy</td>
<td>disease</td>
</tr>
<tr>
<td>1022</td>
<td>healthy</td>
<td>disease</td>
</tr>
</tbody>
</table>
<p>You can get the reads and sequence targets with:</p>
<div class="highlight"><pre><span></span><code>wget https://ngs-longreads-training.s3.eu-central-1.amazonaws.com/groupwork_pacbio.tar.gz
tar -xvf groupwork_pacbio.tar.gz
rm groupwork_pacbio.tar.gz
</code></pre></div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>Download the data file package in your shared working directory, i.e. : <code>/group_work/&lt;group name&gt;</code> or <code>~/&lt;group name&gt;</code>. Only one group member has to do this.</p>
</div>
<p>It has the following directory structure:</p>
<div class="highlight"><pre><span></span><code>groupwork_pacbio
├── alignments
│   ├── bc1020.aln.bam
│   ├── bc1021.aln.bam
│   └── bc1022.aln.bam
├── reads
│   ├── 1015.fastq.gz
│   ├── 1016.fastq.gz
│   ├── 1017.fastq.gz
│   ├── 1018.fastq.gz
│   ├── 1019.fastq.gz
│   ├── 1020.fastq.gz
│   ├── 1021.fastq.gz
│   └── 1022.fastq.gz
└── targets
    ├── target_gene1_hg38.bed
    └── target_gene2_hg38.bed

3 directories, 13 files
</code></pre></div>
<p>The targets in gene1 and gene2 are described in <code>targets/target_gene1_hg38.bed</code> and <code>targets/target_gene2_hg38.bed</code> respectively. The columns in these <code>.bed</code> files describe the chromosome, start, end, name, motifs, and whether the motifs are in reverse complement.</p>
<p>You can download the reference genome like this:</p>
<div class="highlight"><pre><span></span><code><span class="nb">cd</span> groupwork_pacbio
mkdir reference
<span class="nb">cd</span> reference
wget ftp://ftp.ensembl.org/pub/release-101/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz
gunzip Homo_sapiens.GRCh38.dna.primary_assembly.fa.gz
</code></pre></div>
<h3 id="before-you-start_1">Before you start</h3>
<p>You can start this project with dividing initial tasks. Because some intermediate files are already given, participants can develop scripts/analyses at different steps of the full analysis from the start. Possible starting points are:</p>
<ul>
<li>Browse IGV to find the genes</li>
<li>Perform the QC with <code>NanoPlot</code></li>
<li>Perform the alignment with <code>minimap2</code></li>
<li>Do the repeat analysis with <code>makeReports.sh</code></li>
</ul>
<p>Alignment files to do an initial repeat analysis are in the <code>tar.gz</code> package. However, it contains only the files for individuals with disease2. You can develop scripts and analyses based on that. To do the full analysis, all the alignments will need to be run.</p>
<h3 id="tasks-questions_1">Tasks &amp; questions</h3>
<ul>
<li>Load the bed files into IGV and navigate to the regions they annotate.<ul>
<li>In which genes are the targets?</li>
<li>What kind of diseases are associated with these genes?</li>
</ul>
</li>
<li>Perform a quality control with <code>NanoPlot</code>.<ul>
<li>How is the read quality? These are circular concensus sequences (ccs). Is this quality expected?</li>
<li>How is the read length?</li>
</ul>
</li>
<li>Align the reads to hg38 with <code>minimap2</code>. For the option <code>-x</code> you can use <code>asm20</code>. Generate separate alignment files for each individual.</li>
</ul>
<div class="admonition danger">
<p class="admonition-title">Start the alignment as soon as possible</p>
<p>The alignment takes quite some time. Try to start the alignment as soon as possible. You can speed up your alignment by first making an index, e.g.:</p>
<div class="highlight"><pre><span></span><code>minimap2 <span class="se">\</span>
-x asm20 <span class="se">\</span>
-d reference/Homo_sapiens.GRCh38.dna.primary_assembly.fa.mmi <span class="se">\</span>
reference/Homo_sapiens.GRCh38.dna.primary_assembly.fa
</code></pre></div>
<p>Refer to the generated index (<code>.mmi</code> file) as reference in the alignment command, e.g.:</p>
<div class="highlight"><pre><span></span><code>minimap2 \
-a \
-x asm20 \
-t 4 \
reference/Homo_sapiens.GRCh38.dna.primary_assembly.fa.mmi \
reads/&lt;my_reads.fastq.gz&gt;
</code></pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Alternatively use <a href="https://github.com/PacificBiosciences/pbmm2"><code>pbmm2</code></a></p>
<p>Pacific Biosciences has developed a wrapper for <code>minimap2</code> that contains settings specific for PacBio reads, named <a href="https://github.com/PacificBiosciences/pbmm2"><code>pbmm2</code></a>. It might slightly improve your alignments. It is installed in the conda environment. Feel free to give it a try if you have time left.</p>
</div>
<ul>
<li>Clone the PacBio <a href="https://github.com/PacificBiosciences/apps-scripts.git">apps-scripts repository</a> to the server. All dependencies are in the conda environment <code>pacbio</code>. Activate it with <code>conda activate pacbio</code>. The script apps-scripts/RepeatAnalysisTools/makeReports.sh generates repeat expansion reports. Check out the <a href="https://github.com/PacificBiosciences/apps-scripts/tree/master/RepeatAnalysisTools#auto-generate-all-reports">documentation</a>, and generate repeat expansion reports for all individuals on both gene1 and gene2.</li>
<li>Check out the report output and read the further <a href="https://github.com/PacificBiosciences/apps-scripts/tree/master/RepeatAnalysisTools#auto-generate-all-reports">documentation of RepeatAnalysisTools</a>.<ul>
<li>How is the enrichment?</li>
<li>Does the clustering make sense? How does the clustering look in IGV?</li>
<li>Which individual is affected with which disease?</li>
<li>Based on the size of the expansions, can you say something about expected disease severity?</li>
</ul>
</li>
</ul>
<blockquote>
<p>This tutorial is based on data provided by Pacific Biosciences at https://downloads.pacbcloud.com/public/dataset/RepeatExpansionDisorders_NoAmp/</p>
</blockquote>

              
            </article>
          </div>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
    <nav class="md-footer__inner md-grid" aria-label="Footer">
      
        
        <a href="../applications/" class="md-footer__link md-footer__link--prev" aria-label="Previous: Applications" rel="prev">
          <div class="md-footer__button md-icon">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
          </div>
          <div class="md-footer__title">
            <div class="md-ellipsis">
              <span class="md-footer__direction">
                Previous
              </span>
              Applications
            </div>
          </div>
        </a>
      
      
    </nav>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    <script id="__config" type="application/json">{"base": "../..", "features": [], "translations": {"clipboard.copy": "Copy to clipboard", "clipboard.copied": "Copied to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search", "search.result.placeholder": "Type to start searching", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.term.missing": "Missing", "select.version.title": "Select version"}, "search": "../../assets/javascripts/workers/search.bd0b6b67.min.js", "version": {"provider": "mike"}}</script>
    
    
      <script src="../../assets/javascripts/bundle.467223ff.min.js"></script>
      
        <script src="../../javascripts/config.js"></script>
      
        <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
      
        <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
      
    
  </body>
</html>